src/ripple/app/tx/impl/NFTokenAcceptOffer.cpp

```diff
@@ -112,20 +112,44 @@ NFTokenAcceptOffer::preclaim(PreclaimContext const& ctx)
         if ((*so)[sfAmount] > (*bo)[sfAmount])
             return tecINSUFFICIENT_PAYMENT;
 
-        // If the buyer specified a destination, that destination must be
-        // the seller or the broker.
+        // If the buyer specified a destination
         if (auto const dest = bo->at(~sfDestination))
         {
-            if (*dest != so->at(sfOwner) && *dest != ctx.tx[sfAccount])
+            // fixUnburnableNFToken: Disabled
+            // the destination must be the seller or the broker.
+            if (!ctx.view.rules().enabled(fixUnburnableNFToken) &&
+                *dest != so->at(sfOwner) && *dest != ctx.tx[sfAccount])
+            {
                 return tecNFTOKEN_BUY_SELL_MISMATCH;
+            }
+
+            // fixUnburnableNFToken: Enabled
+            // the destination may only be the account brokering the offer
+            if (ctx.view.rules().enabled(fixUnburnableNFToken) &&
+                *dest != ctx.tx[sfAccount])
+            {
+                return tecNO_PERMISSION;
+            }
```

Comment:
it's not a blocker, but you are checking the rule twice when you could check it just once with some conditional re-ordering

Reply:
rewrote it please check it out.

Reply:
it's better looking to me! this is fine for merge, but IMO I would do this to remove a level of indentation:


```c++
if (ctx.view.rules().enabled(fixUnburnableNFToken))
{
  if (*dest != ctx.tx[sfAccount])
    return tecNO_PERMISSION;
}
else if (*dest != so->at(sfOwner) && *dest != ctx.tx[sfAccount])
  return tecNFTOKEN_BUY_SELL_MISMATCH;
```